### PR TITLE
fix: upgraded Jest to v29

### PR DESCRIPTION
**Pull request by KevinGhadyani-Okta**
### Description

Upgraded Jest to v29.

### CODE REVIEW

File: packages/babel-plugin-fully-specified/package.json [hunk 1]
```diff
@@ -21,11 +21,11 @@
     "@babel/types": "^7.14.5"
   },
   "devDependencies": {
-    "@babel/core": "^7.15.0",
+    "@babel/core": "^7.19.1",
     "@okta/odyssey-lifecycle": "^0.15.0",
     "@okta/odyssey-typescript": "^0.15.0",
-    "babel-jest": "^26.6.3",
```
Comment (kamronbatmanghelich-okta) [inline packages/babel-plugin-fully-specified/package.json:27]: To confirm, we don't need babel-jest?

File: packages/odyssey-react/src/utils/focusHandling.ts [hunk 2]
```diff
@@ -40,19 +40,14 @@ function setFocus(elem: OptionalHTMLElement): OptionalHTMLElement {
...
+  const focusableItems = Array.from(allItems).filter((el) => el.tabIndex >= 0);
   // Capture original focused element before setting focus inside modal dialog
   const lastFocusedElement = document.activeElement;
   if (focusableItems.length > 0) {
-    requestAnimationFrame(() => {
+    (() => {
       // Focus on first focusable element inside dialog
       focusableItems[0].focus();
-    });
+    })();
```
Comment (kamronbatmanghelich-okta) [inline packages/odyssey-react/src/utils/focusHandling.ts:50]: Can you explain the preference here between requestAnimationFrame and an immediate execution of the function?

Comment (KevinGhadyani-Okta) [reply packages/odyssey-react/src/utils/focusHandling.ts:50]: Oops. This is a mistake. I shouldn't have changed it. Kept it in by mistake.

File: packages/odyssey-react/package.json [hunk 3]
```diff
@@ -21,7 +21,7 @@
   },
   "dependencies": {
     "@okta/odyssey-react-theme": "^0.15.0",
-    "@react-aria/focus": "3.5.0",
+    "@react-aria/focus": "3.8.0",
```
Comment (kamalgill-okta) [inline packages/odyssey-react/package.json:24]: We no longer need this dependency given we've removed `FocusScope`

**Review by kamalgill-okta — CHANGES_REQUESTED** [review]

File: packages/odyssey-react/src/utils/focusHandling.ts (hunk 2 above)
Comment (KevinGhadyani-Okta) [reply packages/odyssey-react/src/utils/focusHandling.ts]: 3872c7e75ea017781dc8e53526055f232efcbf0b

Fixed

File: packages/odyssey-react/package.json (hunk 3 above)
Comment (KevinGhadyani-Okta) [reply packages/odyssey-react/package.json]: 50687c3c7751d1b48ee71f2b1996e7563ea4bc6e

File: packages/odyssey-react/src/utils/focusHandling.ts [hunk 4]
```diff
@@ -40,12 +40,7 @@ function setFocus(elem: OptionalHTMLElement): OptionalHTMLElement {
   const allItems: NodeListOf<HTMLElement> = elem.querySelectorAll(
     FOCUSABLE_ITEMS_SELECTOR
   );
-  const focusableItems = Array.from(allItems).filter((el) => {
-    if (el.tabIndex === -1) {
-      return false;
-    }
-    return true;
-  });
+  const focusableItems = Array.from(allItems).filter((el) => el.tabIndex >= 0);
```
Comment (kamalgill-okta) [inline packages/odyssey-react/src/utils/focusHandling.ts:43]: Aren't there focusable elements (such as buttons) that don't need an explicit `tabindex` attribute?

**Review by kamalgill-okta — APPROVED** [review]
LGTM 👍

File: packages/odyssey-react/src/utils/focusHandling.ts (hunk 4 above)
Comment (kamalgill-okta) [reply packages/odyssey-react/src/utils/focusHandling.ts:43]: Feel free to ignore this comment given the relevant code wasn't modified in this PR, but we should probably at some point improve the focusableItems lookup to account for elements that are natively focusable without requiring a tabindex attribute.

File: packages/babel-plugin-fully-specified/package.json (hunk 1 above)
Comment (KevinGhadyani-Okta) [reply packages/babel-plugin-fully-specified/package.json:27]: Correct. It's built into Jest.

**Review by kamronbatmanghelich-okta — APPROVED** [review]

**Pull request merged**